tests/frame.test.js

```diff
@@ -0,0 +1,236 @@
+const Frame = require("../main/frame.cjs");
+
+describe("frame", () => {
+  it("has a score property of 0", () => {
+    const frame = new Frame();
+    expect(frame.score).toEqual(0);
+  });
+
+  it('returns the frame', () => {
+    const frame = new Frame();
+    expect(frame.addRoll(5)).toBeInstanceOf(Frame)
+  })
+
+  describe("addRoll", () => {
+    it("adds first roll to frame", () => {
+      const frame = new Frame();
+      frame.addRoll(5);
+      expect(frame.roll1).toEqual(5);
+    });
+
+    it("adds second roll to frame", () => {
+      const frame = new Frame();
+      frame.addRoll(3);
+      frame.addRoll(7);
+      expect(frame.roll1).toEqual(3);
+      expect(frame.roll2).toEqual(7);
+    });
+
+    describe('#validateRoll', () => {
+      describe('when invalid 2nd roll', () => {
+        it('throws an error', () => {
+          const frame = new Frame();
+          frame.addRoll(5)
+          expect(() => { frame.addRoll(6) }).toThrow('invalid roll')
+        })
+      })
+
+      describe('when given 11', () => {
+        it('throws an error', () => {
+          const frame = new Frame();
+          expect(() => {frame.addRoll(11)}).toThrow('invalid roll')
+          
+        })
+      })
+    })
+
+    describe('#summariseFrame', () => {
+      describe('when strike', () => {
+        it('sets roll2 as null', () => {
+          const frame = new Frame();
+          frame.addRoll(10)
+          expect(frame.roll2).toEqual(null)
+        })
+
+        it('assigns 2 bonus points', () => {
+          const frame = new Frame();
+          frame.addRoll(10)
+          expect(frame.bonusPoints).toEqual(2)
+        })
+      })
+
+      describe('when spare', () => {
+        it('assigns 1 bonus points', () => {
+          const frame = new Frame();
+          frame.addRoll(5)
+          frame.addRoll(5)
+          expect(frame.bonusPoints).toEqual(1)
+        })
+      })
+      
+      describe('when open', () => {
+        it('assigns 0 bonus points', () => {
+          const frame = new Frame();
+          frame.addRoll(5)
+          frame.addRoll(4)
+          expect(frame.bonusPoints).toEqual(0)
```

Comment:
Here, you ideally need some way to access the total number of points for this frame — it should be exposed through a method, e.g `frame.getBonusPoints()`, so again, we don't expose too much the internal implementation with attribute names